calico.go

```diff
@@ -306,78 +301,40 @@ func cmdDel(args *skel.CmdArgs) error {
 		"Node":         nodename,
 	}).Info("Extracted identifiers")
 
-	// Always try to release the address. Don't deal with any errors till the endpoints are cleaned up.
-	fmt.Fprintf(os.Stderr, "Calico CNI releasing IP address\n")
-	logger.WithFields(log.Fields{"paths": os.Getenv("CNI_PATH"),
-		"type": conf.IPAM.Type}).Debug("Looking for IPAM plugin in paths")
-
-	// We need to replace "usePodCidr" with a valid, but dummy podCidr string with "host-local" IPAM.
-	if conf.IPAM.Type == "host-local" && strings.EqualFold(conf.IPAM.Subnet, "usePodCidr") {
-
-		// host-local IPAM releases the IP by ContainerID, so podCidr isn't really used to release the IP.
-		// It just needs a valid CIDR, but it doesn't have to be the CIDR associated with the host.
-		dummyPodCidr := "0.0.0.0/0"
-		var stdinData map[string]interface{}
-
-		if err := json.Unmarshal(args.StdinData, &stdinData); err != nil {
-			return err
-		}
-
-		logger.WithField("podCidr", dummyPodCidr).Info("Using a dummy podCidr to release the IP")
-		stdinData["ipam"].(map[string]interface{})["subnet"] = dummyPodCidr
-
-		args.StdinData, err = json.Marshal(stdinData)
-		if err != nil {
-			return err
-		}
-		logger.WithField("stdin", args.StdinData).Debug("Updated stdin data for Delete Cmd")
-	}
-
-	ipamErr := ipam.ExecDel(conf.IPAM.Type, args.StdinData)
-
-	if ipamErr != nil {
-		logger.Error(ipamErr)
-	}
-
 	calicoClient, err := CreateClient(conf)
 	if err != nil {
 		return err
 	}
 
-	ep := api.WorkloadEndpointMetadata{
+	wep := api.WorkloadEndpointMetadata{
 		Name:         args.IfName,
 		Node:         nodename,
 		Orchestrator: orchestrator,
-		Workload:     workload}
-	if err = calicoClient.WorkloadEndpoints().Delete(ep); err != nil {
+		Workload:     workload,
+	}
+
+	// Handle k8s specific bits of handling the DEL.
+	if orchestrator == "k8s" {
+		return k8s.CmdDelK8s(calicoClient, wep, args, conf, logger)
+	}
+
```

Comment:
that is because if the deleting WEP fails we return an error so DEL operation doesn't go through, I thought moving IPAM deletion after makes sense, but I guess it makes more sense in k8s.go not so much here